chain-signatures/contract/src/lib.rs

```diff
@@ -117,6 +117,8 @@ impl MpcContract {
     fn add_sign_result(&mut self, payload: &SignatureRequest, signature: SignatureResponse) {
         if self.pending_requests.contains_key(payload) {
             self.pending_requests.insert(payload, &Some(signature));
+        } else {
+            env::panic_str(&format!("Key not found: {:?}", payload))
```

Comment:
This needs a test, I'll add it in

Reply:
if it's discarded, we need to alert the requestor of the signature that it failed to publish somehow. I guess with yield/resume we can do this by returning an error and pancing so we can do this in future, but maybe for now we just store the signature to be sent again on the next iteration

Reply:
For now we can add a record_failure method to contract, and cross contract call to it when it fails. The record_failure can be actually not record anything on chain, but because it is cross contract called, we can modify the indexer part to catch it.

Of course yield/resume would be better if it can be implemented soon, and signature discarded as a short term solution is acceptable.

Reply:
I ended up pushing it to the back of the queue and retrying unless it fails in the contract. It's easy to DoS the contract if you allow incorrect responses to cancel the request.